variables.tf

```diff
@@ -183,3 +182,34 @@ variable "disable_mfa" {
   type        = bool
   default     = false
 }
+
+variable "disabled" {
+  description = "Specifies whether the user is disabled, which prevents logging in and aborts all the currently-running queries for the user."
+  type        = bool
+  default     = false
+}
+
+variable "naming_scheme" {
+  description = <<EOT
+  Naming scheme configuration for the resource. This configuration is used to generate names using context provider:
+    - `properties` - list of properties to use when creating the name - is superseded by `var.context_templates`
+    - `delimiter` - delimited used to create the name from `properties` - is superseded by `var.context_templates`
+    - `context_template_name` - name of the context template used to create the name
+    - `replace_chars_regex` - regex to use for replacing characters in labels created by the provider - any characters that match the regex will be removed from the name
+    - `extra_labels` - map of extra label-value pairs, used to create a name
+  EOT
+  type = object({
+    properties            = optional(list(string), ["environment", "name"])
+    delimiter             = optional(string, "_")
+    context_template_name = optional(string, "snowflake-user")
+    replace_chars_regex   = optional(string, "[^a-zA-Z0-9_]")
+    extra_labels          = optional(map(string))
```

Comment:
we have `properties` and `labels` here. Frankly saying they are very related. 
Perhaps `extra_properties_map` or since in the data source we pass it to the `values` option, they could be `extra_values`?

Reply:
I had a problem with this one also...
`extra_values` doesn't really say anything
`extra_properties` also
`extra_property_values` is the most informative, but in the same time seems long and had to understand by outsider
That's why I came with `extra_labels` - but in the same time it might conflict with GCP tagging `labels` :thinking:

Reply:
Since there is no ideal candidate, it was decided to go with `extra_values` since that is an additional set of `values` passed to `context` provider